modules/images/webp-uploads/image-edit.php

```diff
@@ -129,16 +166,38 @@ function ( $metadata, $post_meta_id ) use ( $post_id, $file_path, $mime_type, $e
 					continue;
 				}
 
-				$extension   = explode( '|', $allowed_mimes[ $targeted_mime ] );
-				$destination = trailingslashit( $original_directory ) . "{$filename}.{$extension[0]}";
-				$result      = $editor->save( $destination, $targeted_mime );
-
-				if ( is_wp_error( $result ) ) {
-					continue;
+				$extension = explode( '|', $allowed_mimes[ $targeted_mime ] );
+				$extension = $extension[0];
+
+				if ( 'thumbnail' === $target ) {
+					if ( ! isset( $subsized_images[ $original_mime_type ]['thumbnail']['file'] ) ) {
+						continue;
+					}
+					$thumbnail_file = $subsized_images[ $original_mime_type ]['thumbnail']['file'];
+					$image_path     = path_join( $original_directory, $thumbnail_file );
+					$editor         = wp_get_image_editor( $image_path, array( 'mime_type' => $targeted_mime ) );
+
+					if ( is_wp_error( $editor ) ) {
+						continue;
+					}
+
+					$current_extension = pathinfo( $thumbnail_file, PATHINFO_EXTENSION );
+					// Create a file with then new extension out of the targeted file.
+					$target_file_name                  = preg_replace( "/\.$current_extension$/", ".$extension", $thumbnail_file );
+					$target_file_location              = path_join( $original_directory, $target_file_name );
+					$result                            = $editor->save( $target_file_location, $targeted_mime );
+					$subsized_images[ $targeted_mime ] = array( 'thumbnail' => $result );
```

Comment:
`$result` can be a `WP_Error`, so we need to check that before using it, similar to how it's already in the `else` clause below.

Reply:
Fixed this!